mkosi

```diff
@@ -2204,42 +2259,125 @@ def install_arch(args: CommandLineArguments, root: str, do_run_build_script: boo
                     )
                 )
 
-    def stop_gpg_agent() -> None:
-        # Kill the gpg-agent used by pacman and pacman-key
-        run(['gpgconf', '--homedir', os.path.join(root, 'etc/pacman.d/gnupg'), '--kill', 'all'])
+    if not do_run_build_script and args.bootable:
+        hooks_dir = os.path.join(root, "etc/pacman.d/hooks")
+        scripts_dir = os.path.join(root, "etc/pacman.d/scripts")
 
-    def run_pacman_key(args: List[str]) -> CompletedProcess:
-        cmdline = [
-            "pacman-key",
-            "--nocolor",
-            "--config", pacman_conf,
-        ]
+        os.makedirs(hooks_dir, 0o755, exist_ok=True)
+        os.makedirs(scripts_dir, 0o755, exist_ok=True)
 
-        try:
-            return run(cmdline + args, check=True)
-        finally:
-            stop_gpg_agent()
+        # Disable depmod pacman hook as depmod is handled by kernel-install as well.
+        os.symlink("/dev/null", os.path.join(hooks_dir, "60-depmod.hook"))
+
+        kernel_add_hook = os.path.join(hooks_dir, "mkosi-kernel-add.hook")
```

Comment:
This hook should also get a numerical prefix, e.g. `90-`, since this is the number of the appropriate mkinitcpio hook.